src/tools/Avalonia.Generators/Common/Domain/IViewResolver.cs

```diff
@@ -1,11 +1,45 @@
+using System.Collections.Immutable;
 using XamlX.Ast;
-using XamlX.TypeSystem;
 
 namespace Avalonia.Generators.Common.Domain;
 
 internal interface IViewResolver
 {
-    ResolvedView? ResolveView(string xaml);
+    ResolvedViewDocument? ResolveView(string xaml);
 }
 
-internal record ResolvedView(string ClassName, IXamlType XamlType, string Namespace, XamlDocument Xaml);
+internal record ResolvedViewInfo(string ClassName, string Namespace)
+{
+    public string FullName => $"{Namespace}.{ClassName}";
+    public override string ToString() => FullName;
+}
+
+internal record ResolvedViewDocument(string ClassName, string Namespace, XamlDocument Xaml)
+    : ResolvedViewInfo(ClassName, Namespace);
+
+internal record ResolvedXmlView(
+    string ClassName,
+    string Namespace,
+    ImmutableArray<ResolvedXmlName> XmlNames)
+    : ResolvedViewInfo(ClassName, Namespace)
+{
+    public ResolvedXmlView(ResolvedViewInfo info, ImmutableArray<ResolvedXmlName> xmlNames)
+        : this(info.ClassName, info.Namespace, xmlNames)
+    {
+        
+    }
+}
+
+internal record ResolvedView(
+    string ClassName,
+    string Namespace,
+    bool IsWindow,
+    ImmutableArray<ResolvedName> Names)
```

Comment:
`ImmutableArray` equality is comparing the underlying array instead of its items so caching won't work. This needs a wrapper type.